numpy/ma/core.pyi

```diff
@@ -513,9 +513,9 @@ class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
     @mask.setter
     def mask(self, value: _ArrayLikeBool_co, /) -> None: ...
     @property
-    def recordmask(self): ...
+    def recordmask(self) -> NDArray[MaskType] | MaskType: ...
```

Comment:
I've set this the same as `mask`

https://github.com/numpy/numpy/blob/8b2321d03c1abce671a15f9d08a97249bd2d161d/numpy/ma/core.pyi#L511-L512

Reply:
if it returns an array, does the returned array shape match that of the masked array?

Reply:
As far as I can tell - yes!

If it's a structured array, then the `recordmask` is no longer a structured array, but it still preserves the shape, whereas `mask` would return a structured array for the mask (and, also, preserve the shape)